feature-libs/product-configurator/rulebased/components/attribute/composition/configurator-attribute-composition.directive.ts

```diff
@@ -10,40 +10,74 @@ import {
   Injector,
   Input,
   isDevMode,
+  OnChanges,
   OnInit,
   ViewContainerRef,
 } from '@angular/core';
-import { LoggerService } from '@spartacus/core';
-import { ConfiguratorAttributeCompositionConfig } from './configurator-attribute-composition.config';
+import {
+  FeatureConfigService,
+  LoggerService,
+  ObjectComparisonUtils,
+} from '@spartacus/core';
+import {
+  AttributeComponentAssignment,
+  ConfiguratorAttributeCompositionConfig,
+} from './configurator-attribute-composition.config';
 import { ConfiguratorAttributeCompositionContext } from './configurator-attribute-composition.model';
+import { Configurator } from '../../../core/model/configurator.model';
 
 @Directive({
   selector: '[cxConfiguratorAttributeComponent]',
 })
-export class ConfiguratorAttributeCompositionDirective implements OnInit {
+export class ConfiguratorAttributeCompositionDirective
+  implements OnInit, OnChanges
+{
   @Input('cxConfiguratorAttributeComponent')
   context: ConfiguratorAttributeCompositionContext;
 
+  protected lastRenderedAttribute: Configurator.Attribute;
+
   protected logger = inject(LoggerService);
+  protected featureConfig = inject(FeatureConfigService);
+
+  protected readonly attrCompAssignment: AttributeComponentAssignment =
```

Comment:
Could you add a missing test? 
![image](https://github.com/SAP/spartacus/assets/61147963/ba2bfcf7-cd6e-4953-a69c-4593694205fb)